impeller/entity/geometry.cc

```diff
@@ -695,4 +699,54 @@ std::optional<Rect> CoverGeometry::GetCoverage(const Matrix& transform) const {
   return Rect::MakeMaximum();
 }
 
+/////// Rect Geometry ///////
+
+RectGeometry::RectGeometry(Rect rect) : rect_(rect) {}
+
+RectGeometry::~RectGeometry() = default;
+
+GeometryResult RectGeometry::GetPositionBuffer(const ContentContext& renderer,
+                                               const Entity& entity,
+                                               RenderPass& pass) {
+  constexpr uint16_t kRectIndicies[4] = {0, 1, 2, 3};
+  auto& host_buffer = pass.GetTransientsBuffer();
+  return GeometryResult{
+      .type = PrimitiveType::kTriangleStrip,
+      .vertex_buffer = {.vertex_buffer = host_buffer.Emplace(
+                            rect_.GetPoints().data(), 8 * sizeof(float),
+                            alignof(float)),
+                        .index_buffer = host_buffer.Emplace(
+                            kRectIndicies, 4 * sizeof(uint16_t),
+                            alignof(uint16_t)),
+                        .index_count = 4,
+                        .index_type = IndexType::k16bit},
+      .prevent_overdraw = false,
+  };
+}
+
+GeometryResult RectGeometry::GetPositionColorBuffer(
+    const ContentContext& renderer,
+    const Entity& entity,
+    RenderPass& pass,
+    Color paint_color,
+    BlendMode blend_mode) {
+  // TODO(jonahwilliams): support per-color vertex in rect geometry.
+  return {};
+}
+
+GeometryResult RectGeometry::GetPositionUVBuffer(const ContentContext& renderer,
+                                                 const Entity& entity,
+                                                 RenderPass& pass) {
+  // TODO(jonahwilliams): support texture coordinates in rect geometry.
```

Comment:
nit: file a bug and explain what this would block/where it would go wrong.

Reply:
Its not really a useful API yet. I was actually thinking we should create a specific drawVertices subclass that adds these methods, since none of the other geometries will use it